OpenRA.Mods.Common/AI/BaseBuilder.cs

```diff
@@ -37,10 +40,19 @@ public BaseBuilder(HackyAI ai, string category, Player p, PowerManager pm, Playe
 			playerPower = pm;
 			playerResources = pr;
 			this.category = category;
+			waterAvailable = true;
+			failCount = 0;
+			failRetryTicks = ai.Info.StructureProductionResumeDelay;
 		}
 
 		public void Tick()
 		{
+			// If failed to place something N consecutive times, wait N ticks until resuming building production
+			if (failCount >= ai.Info.MaximumFailedPlacementAttempts && --failRetryTicks > 0)
+				--failRetryTicks;
```

Comment:
Do you really intend on decrementing `failRetryTicks` twice per tick?  You probably want to drop the `--` inside the if check.